setup/system.sh

```diff
@@ -4,6 +4,25 @@ source setup/functions.sh # load our functions
 # Basic System Configuration
 # -------------------------
 
+# ### Ensure system defaults access rights are correctly configured
+
+# If the /etc/default directory has group write rights, the installer will
+# display a lot of warnings during setup
+
+chmod g-w /etc/default
+
+# ### Set hostname of the box
+
+# If the hostname is not correctly resolvable sudo can't be used. This will result in
+# errors during the install
+#
+# First set the hostname in the configuration file, then activate the setting
+# Also make sure that the loopback device resolves to the hostname
+
+echo $PRIMARY_HOSTNAME > /etc/hostname
+hostname $PRIMARY_HOSTNAME
+sed -i "s/127\.0\.1\.1.*/127.0.1.1\t$PRIMARY_HOSTNAME/" /etc/hosts
```

Comment:
https://www.debian.org/doc/manuals/debian-reference/ch05.en.html#_the_hostname_resolution

It is inherited from debian. 

I did this to resolve a problem with sudo, whenever you run sudo without that you get, sudo: unable to resolve host *

http://askubuntu.com/questions/59458/error-message-when-i-run-sudo-unable-to-resolve-host-none

There are some issues with Scaleway, their images are open source. But I think it might be good if we try to make a fool proof installer that corrects most common issues. (Or at least detect them)